packages/itwin/grouping-mapping-widget/src/components/Groups/Editing/GroupAction.tsx

```diff
@@ -197,7 +197,7 @@ export const GroupAction = (props: GroupActionProps) => {
       </div>
       <div className="gmw-action-panel">
         {isLoading && <LoadingSpinner />}
-        {isQueryBuilderStep && <QueryBuilderActionPanel onClickNext={onClickNext} />}
+        {isQueryBuilderStep && <QueryBuilderActionPanel onClickNext={onClickNext} isNextDisabled={!queryRowCount} />}
```

Comment:
This will leave the `Next` button disabled even when all required fields are filled but the query returns 0 results. Query that results 0 results is a valid query. `!0 === true` in js.